dodola/tests/test_services.py

```diff
@@ -351,6 +352,47 @@ def test_remove_leapdays():
     assert len(ds_leapyear.time) == 365
 
 
+@pytest.mark.parametrize(
+    "process", [pytest.param("pre-process"), pytest.param("post-process")]
+)
+def test_correct_wet_day_frequency(process):
+    """Test that wet day frequency correction corrects the frequency of wet days"""
+    # Make some fake precip data
+    n = 1500
+    threshold = 0.05
+    ts = np.linspace(0.0, 10, num=n)
+    ds_precip = _datafactory(ts, start_time="1950-01-01")
+    in_url = "memory://test_correct_wet_day_frequency/an/input/path.zarr"
+    out_url = "memory://test_correct_wet_day_frequency/an/output/path.zarr"
+    repository.write(in_url, ds_precip)
+
+    correct_wet_day_frequency(in_url, out=out_url, process=process)
+    ds_precip_corrected = repository.read(out_url)
+
+    if process == "pre-process":
+        # all 0 values should have been set to a random uniform value below 0.05
+        assert (
+            ds_precip_corrected["fakevariable"].where(
+                ds_precip["fakevariable"] == 0, drop=True
+            )
+            != 0.0
+        )
```

Comment:
Should these `assert`s have `.all()` ? Same for lines L380-393
```suggestion
        assert ((
            ds_precip_corrected["fakevariable"].where(
                ds_precip["fakevariable"] == 0, drop=True
            )
            != 0.0).all()
        )
```

Reply:
Because the test time series only has one 0, that's why I didn't include the `.all()`s here